torch_geometric/nn/models/label_usage.py

```diff
@@ -0,0 +1,92 @@
+import torch
+import torch.nn.functional as F
+
+from torch_geometric.typing import OptTensor, Adj
+
+class LabelUsage(torch.nn.Module):
+    r"""The label usage operator for semi-supervised node classification,
+    as introduced in `"Bag of Tricks for Node Classification"
+    <https://arxiv.org/abs/2103.13355>`_ paper.
+
+    .. note::
+
+        When using the :class:`LabelUsage`, adjust the model's input dimension
+        accordingly to include both features and classes.
+
+    Args:
+        base_model: An instance of the model that will do the 
+            inner forward pass.
+        num_classes (int): Number of classes in dataset
+        split_ratio (float): Proportion of true labels to use as features 
+            during training (default: :obj:'0.5')
+        num_recycling_iterations (int): Number of iterations for the
+            label reuse procedure to cycle predicted soft labels 
+            (default: :obj:'100')
+        return_tuple (bool): If true, returns (output, train_label_idx, 
+            train_pred_idx) otherwise returns prediction output 
+            (default :obj:'False')
+    """
+
+    def __init__(
+        self,
+        base_model: torch.nn.Module,
+        num_classes: int,
+        split_ratio: float = 0.5,
+        num_recycling_iterations: int = 100,
+        return_tuple: bool = False,
+    ):
+
+        super(LabelUsage, self).__init__()
+        self.base_model = base_model
+        self.num_classes = num_classes
+        self.split_ratio = split_ratio
+        self.num_recycling_iterations = num_recycling_iterations
+        self.return_tuple = return_tuple
+
+    def forward(
+        self,
+        x: Tensor, 
+        edge_index: Adj, 
+        y: Tensor, 
+        train_idx: Tensor,
+        batch: OptTensor = None,
+        ):
+        r"""
+        Forward pass using label usage algorithm.
+
+        Args:
+            x (torch.Tensor): Node feature tensor 
+            edge_index (torch.Tensor or SparseTensor): The edge indices
+            y (torch.Tensor): Node label tensor 
```

Comment:
We should be explicit about the expected dimensions.  It looks like it's currently `(N,)` where N is the length of the first dimension of `x`